notebooks/tests/__init__.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 Dremio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Unit tests for demo notebooks."""
+import os
+import subprocess
+import tarfile
+import shutil
+import site
+from typing import Optional
+
+import requests
+
+
+def _find_notebook(notebook_file: str) -> str:
+    path_to_notebook = os.path.join("notebooks", notebook_file)
+    if not os.path.exists(path_to_notebook):
+        path_to_notebook = os.path.join("..", path_to_notebook)
+    if not os.path.exists(path_to_notebook):
+        path_to_notebook = os.path.join("..", path_to_notebook)
+    if not os.path.exists(path_to_notebook):
+        raise Exception(
+            f"Could not find {notebook_file} in {os.path.abspath('.')} and {os.path.abspath('..')}"
+        )
+    return os.path.abspath(path_to_notebook)
+
+
+def _get_unzip(filename: str, url: str) -> None:
+    if not os.path.exists(filename):
+        response = requests.get(url, stream=True)
+        file = tarfile.open(fileobj=response.raw, mode="r|gz")
+        file.extractall(path=".")
+
+
+def _get_spark() -> None:
+    filename = 'spark-3.0.3-bin-hadoop2.7'
+    url = "https://downloads.apache.org/spark/spark-3.0.3/spark-3.0.3-bin-hadoop2.7.tgz"
```

Comment:
fyi: it would be better/more stable to use the Archive URL for Hadoop/Spark (similar to https://github.com/projectnessie/nessie-demos/commit/aa5922cc49a7901a83fbe322ead4473ff92d83a9)

Reply:
done